packages/astro/src/vite-plugins/core.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * A plugin that let users write components that can access the internals
+ * of TutorialKit
+ *
+ * The virtual module can be imported as:
+ *
+ * ```ts
+ * import { webcontainer } from 'tutorialkit:core';
+ *
+ * export const MyComponent = () => {
+ *   useEffect(() => {
+ *     (async () => {
+ *       const webcontainerInstance = await webcontainer;
+ *       await webcontainerInstance.fs.writeFile('foo.js', 'console.log("hello world")');
+ *     })();
+ *   }, []);
+ *
+ *   return null;
+ * };
+ * ```
+ */
+import type { VitePlugin } from '../types.js';
+import { fileURLToPath } from 'node:url';
+import path from 'node:path';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const virtualModuleId = 'tutorialkit:core';
+const resolvedVirtualModuleId = `${virtualModuleId}`;
+
+export const tutorialkitCore: VitePlugin = {
+  name: 'tutorialkit-core-virtual-mod-plugin',
+  resolveId(id) {
+    if (id === virtualModuleId) {
+      return resolvedVirtualModuleId;
+    }
+  },
+  async load(id) {
+    if (id === resolvedVirtualModuleId) {
+      const pathToInit = path.join(__dirname, 'default', 'components', 'webcontainer.ts');
```

Comment:
I would just go with

```suggestion
      const pathToInit = path.join(__dirname, 'default/components/webcontainer.ts');
```